File: 0169-majority-element/0169-majority-element.c
int majorityElement(int* nums, int numsSize) {
    int m = nums[0],
        c = 1;
    for (int i = 1; i < numsSize; i++) {
        if (c == 0) {
            m = nums[i];
            c++;
        } else if (m == nums[i]) {
            c++;
        } else {
            c--;
        }
    }
    return m;
}

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        Map<Integer,Integer> s = new HashMap<>();
        int i;
        for(i=0;i<nums.length;i++)
        {
            s.put(nums[i],s.getOrDefault(nums[i],0)+1);
        }
        int n = nums.length;
        int max=0;
        for(i=0;i<nums.length;i++)
        {
            if(s.get(nums[i])>(n/2))
            {
             max = nums[i];
            }
        }
        return max;
    }
}